packages/web5/lib/src/dids/did_core/did_resolution_result.dart

```diff
@@ -61,13 +61,36 @@ class DidResolutionResult {
             didResolutionMetadata ?? DidResolutionMetadata(),
         didDocumentMetadata = didDocumentMetadata ?? DidDocumentMetadata();
 
-  /// A convenience constructor for creating a [DidResolutionResult] representing
+  /// A factory constructor for creating a [DidResolutionResult] representing
   /// an invalid DID scenario. This sets the resolution metadata error to 'invalidDid'
   /// and leaves the DID document as `null`.
-  DidResolutionResult.invalidDid()
-      : didResolutionMetadata = DidResolutionMetadata(error: 'invalidDid'),
-        didDocument = null,
-        didDocumentMetadata = DidDocumentMetadata();
+  factory DidResolutionResult.invalidDid() {
+    return DidResolutionResult(
+      didResolutionMetadata: DidResolutionMetadata(error: 'invalidDid'),
+      didDocument: null,
+      didDocumentMetadata: DidDocumentMetadata(),
+    );
+  }
+
+  factory DidResolutionResult.notFound() {
+    return DidResolutionResult(
+      didResolutionMetadata: DidResolutionMetadata(
+        error: 'notFound',
+      ),
+      didDocument: null,
+      didDocumentMetadata: DidDocumentMetadata(),
+    );
+  }
+
+  factory DidResolutionResult.representationNotSupported() {
+    return DidResolutionResult(
+      didResolutionMetadata: DidResolutionMetadata(
+        error: 'representationNotSupported',
+      ),
+      didDocument: null,
+      didDocumentMetadata: DidDocumentMetadata(),
+    );
+  }
```

Comment:
nice @StonePack ! what are your thoughts on creating a factory function `DidResolutionResultWithError(ResolutionError err)` where `ResolutionError` is an enum for all of [these](https://www.w3.org/TR/did-spec-registries/#error). This way we don't end up with _n_ of these factories

Reply:
@mistermoe All for it, I'll make that change